clients/src/test/java/org/apache/kafka/common/network/NetworkTestUtils.java

```diff
@@ -66,6 +63,14 @@ public static void checkClientConnection(Selector selector, String node, int min
         }
     }
 
+    public static void waitForChannelReady(Selector selector, String node) throws IOException {
+        // wait for handshake to finish
+        int secondsLeft = 30;
+        while (!selector.isChannelReady(node) && secondsLeft-- > 0) {
+            selector.poll(1000L);
+        }
```

Comment:
Shouldn't we check that the channel is ready once we exit the while loop and if it is not, throw an exception?